internal/auth/tokens.go

```diff
@@ -57,5 +60,18 @@ func LoadTenantToken() (*Token, error) {
 		return nil, err
 	}
 
+	claims := jwt.RegisteredClaims{}
+	if _, _, err := new(jwt.Parser).ParseUnverified(token.TenantToken, claims); err != nil {
+		return nil, err
+	}
+
+	// If stored token is expired, we remove it and return [fs.ErrNotExist].
+	if !claims.VerifyExpiresAt(time.Now(), true) {
```

Comment:
Wondering if I should have some leeway here?

Reply:
You mean throw away a token that is still valid, but is about to expire? Would make sense.

But starting simple with a strict check is fine.

(Not the other way of course - we shouldn't try to use recently expired tokens).

Reply:
> You mean throw away a token that is still valid, but is about to expire? Would make sense.
Yep, that is what I mean.